server/datastore/mysql/hosts.go

```diff
@@ -717,6 +717,47 @@ func shouldCleanTeamPolicies(currentTeamID, newTeamID *uint) bool {
 	return *currentTeamID != *newTeamID
 }
 
+func (ds *Datastore) EnrollOrbit(ctx context.Context, hardwareUUID string, orbitNodeKey string) (*fleet.Host, error) {
+	if orbitNodeKey == "" {
+		return nil, ctxerr.New(ctx, "orbit node key is empty")
+	}
+
+	if hardwareUUID == "" {
+		return nil, ctxerr.New(ctx, "harware uuid is empty")
+	}
+
+	var host fleet.Host
+	err := ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		var hostID int64
+
+		err := sqlx.GetContext(ctx, tx, &host, `SELECT id FROM hosts WHERE uuid = ?`, hardwareUUID)
+		switch {
+		case err != nil && !errors.Is(err, sql.ErrNoRows):
+			return ctxerr.Wrap(ctx, err, "some error")
+		case errors.Is(err, sql.ErrNoRows):
```

Comment:
Case statements are evaluated in order, so if you move this to the top then the second case clause can just be `err != nil`